test/unit/basic.dev.test.js

```diff
@@ -60,10 +62,11 @@ describe('basic dev', () => {
 
   test('Config: build.transpile', () => {
     expect(transpile('vue-test')).toBe(true)
-    expect(transpile('node_modules/test.js')).toBe(false)
-    expect(transpile('node_modules/vue-test')).toBe(true)
-    expect(transpile('node_modules/vue.test.js')).toBe(true)
-    expect(transpile('node_modules/test.vue.js')).toBe(true)
+    expect(transpile(path.normalize('node_modules/test.js'))).toBe(false)
+    expect(transpile(path.normalize('node_modules/vue-test'))).toBe(true)
+    expect(transpile(path.normalize('node_modules/vue.test.js'))).toBe(true)
+    expect(transpile(path.normalize('node_modules/test.vue.js'))).toBe(true)
+    expect(transpile(path.normalize('node_modules/@scoped/packageA/src/index.js'))).toBe(true)
```

Comment:
We should also mock windows path, like 'node_modules\\\\@scoped\\\\packageA\\\\src\\\\index.js'

Reply:
Hi @clarkdo, I am not sure how I can mock windows path here and still have the test run correctly on both POSIX and Windows due to the single direction POSIX to Windows normalization by `path.normalize`.

Also, in all case (I suspect?) babel will always pass native filepath into the exclude function automatically.

Reply:
> Also, in all case (I suspect?) babel will always pass native filepath into the exclude function automatically.

I meant to imply that: Since native filepath is used, we should be able to cover both case for 
Windows and POSIX using (ci.azure) and (ci/circleci: test-unit).

Is that correct?

Reply:
Actually not, exclude file path will be `c:\\project\\node_modules\\@Scoped\\packageA\\src\\index.js`.

So we may also need to normalize `file` in `exclude`.

That's also why I mentioned we'd better to mock windows path, like:

```js
transpile('node_modules\\\\@scope\\\\module')
```

Reply:
If we don't consider compatibility, we can only normalize `file` in `exclude`.

Reply:
Hi @clarkdo, sorry I may not have understood on what you meant by compatibility, I am still unclear about why we need to normalize the filepath (`file`) in the `exclude` function.

## Clarifications:
1. Just to be clear, when we talk about `normalize`, this is what we mean right?
    > to convert a given file path to a new plain string with the OS's separator
2. I do understand that the filepath in the exclude function will be `"c:\\project\\node_modules\\@scoped\\packageA\\src\\index.js"` in Windows.
However, I am not sure about why we should do `transpile('node_modules\\\\@scope\\\\module')` in the test.
    1. Firstly, this will always fail the test for POSIX system, since our entry in options.build.transpile will be normalized to test for `/`.

    2. This will always fail for Windows too since we are testing `/@scope\\module/` against `"@scope\\\\module"`
    because:
    `/@scope\\module/.test("@scope\\\\module") === false`

    3. Using 4 backslash seems to imply that you want to convert the string to a RegExp? i.e. 
    `new RegExp('@scope\\\\module').source === /@scope\\module/.source`

    Therefore, the more accurate behavior will be to do `transpile(path.normalize('node_modules/@scope/module'))` instead? as this will create the same behavior as how webpack passes filepath to the `exclude` function.




## My thoughts:
https://github.com/nuxt/nuxt.js/blob/3702dfe5dcb5c4fdeb03fcd3025f5a5ae39232be/packages/webpack/src/config/base.js#L173-L176

According to my understanding, `exclude` function will receive a `filepath` as parameter. The file path will be constructed depending on the OS. On Windows it will always be separated with `\` on POSIX it will be seperated with `/`.

`modulesToTranspile` will contain RegExps using the file path we specified in the `options.build.transpile` array. These RegExps' file path will be normalize according to the OS.

Since e85a201, regardless of the format that user specify (POSIX or Windows) they will always get normalized to the OS's format. This will match well with the logic in exclude function.

## Illustration of logic:
#### Windows

| input                | moduleToTranspile    | exclude's param |
| :--                  | :--                  | :--             |
| `"@scoped/packageA"`  | `/@scoped\\packageA/` | `"c:\\project\\node_modules\\@scoped\\packageA\\src\\index.js"`
| `"@scoped\\packageA"` | `/@scoped\\packageA/` | `"c:\\project\\node_modules\\@scoped\\packageA\\src\\index.js"`

#### POSIX

| input                | moduleToTranspile    | exclude's param |
| :--                  | :--                  | :--             |
| `"@scoped/packageA"`  | `/@scoped\/package/` | `"/project/node_modules/@scoped/packageA/src/index.js"`
| `"@scoped\\packageA"` | `/@scoped\/package/` | `"/project/node_modules/@scoped/packageA/src/index.js"`

## Tested
Both of these cases are written as part of the test case too
https://github.com/nuxt/nuxt.js/blob/e85a201be01ca8dcd8e59f4fcc326cf2b501b323/test/unit/basic.dev.test.js#L29-L34

and both passed the unit test ran in Windows (in [ci.azure](https://dev.azure.com/nuxt/nuxt.js/_build/results?buildId=612&view=logs)) and POSIX environment (in [circleci](https://circleci.com/gh/nuxt/nuxt.js/15562?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link))

---

Thank for you reading through my long comment. :)

Reply:
Thanks you the clarifications, it works fine after converting POSIX commit.

The `\\\\` is because github will escape the `\` in quote, so I add duplicate ones in whole comment, sorry.